syntax = "proto3";

package voyager;

enum ErrorCode {
  ERROR_CODE_OK = 0;
  ERROR_CODE_TIMEOUT = 1;
  ERROR_CODE_INVALID_SERVICE = 2;
  ERROR_CODE_INVALID_METHOD =3;
  ERROR_CODE_INVALID_REQUEST = 4;
  ERROR_CODE_UNKNOWN = 5;
}

message RpcMessage {
  int32 id = 1;
  string service_name = 2;
  string method_name = 3;
  bytes data = 4;
  ErrorCode error = 5;
}
